### ISSUE TITLE
Default "EVALUATE" to the first line of the Query Editor, or a comment line, then EVALUATE

### ISSUE THREAD

**Issue by TRTruax422** [DaxStudio/DaxStudio#454]
Given the paucity of help in DAX Studio, many users may not know about the requirement to use EVALUATE in the Query Editor (myself included, obviously).

**Comment by marcosqlbi** [comment]
I'm worried that this would only shift the issue.
If you copy a scalar expression like "SUM ( Sales[Quantity] )" in DAX Studio, you would need this code to execute it:
```
EVALUATE {
    SUM ( Sales[Quantity] )
}
```

However, a table expression like Store does not require the table constructor syntax:
```
EVALUATE 
    Store
```
 
Finally, the definition of a measure would not work if not included in a larger query. If you write:
```
SUMX ( Sales, Sales[Quantity] * Sales[Net Price] )
```
you should execute:
```
DEFINE MEASURE Sales[Sales Amount] = 
    SUMX ( Sales, Sales[Quantity] * Sales[Net Price] )
EVALUATE 
    Store
```
The latter case is the more complex to interpret, because the assignment could be confused with a Boolean expression, which could be executed by the first syntax (even though usually the left part of the assignment is not a valid scalar expression).

There could be a comment, or an expression that default to EVALUATE followed by the name of a table of the model (but which one? A large one would create a performance issue if executed). Such a behavior should be activated only in new installation and the user should be able to "disable" it in the configuration.
However, I'm always worried to provide a suggestion that could move users in the wrong direction. The really necessary step is to learn the DAX query syntax, the Query Builder can provide help here, maybe there should be an offer to activate the Query Builder for new users.

**Comment by dgosbell** [comment]
Actually I've been thinking of maybe adding a getting started "watermark" to the main editor window that would show up if there is no text in the editor and we could probably also then hyperlink off to a page on our website to give more detailed examples of DAX queries. I think this might work better than injecting default text.

**Comment by marcosqlbi** [comment]
Yes I agree!

**Comment by dgosbell** [comment]
This should be fixed in the 2.14 release which was just published. We've now added help text when the main code pain is empty which should help new users figure out how to get started.

**Comment by TRTruax422** [comment]
Darren,

I downloaded version 2.14 last week, and think the text and link are helpful. I also found all of Alberto’s DAX Studio videos, and they were very helpful. I think it would also be helpful to add a link to SQLBI’s YouTube channel in the help menu pull-down.

Also, making Query Builder visible by default is a real plus. I did not know about that option until I saw the video about it.

Tim
